### PR TITLE
Use AttributeValue in Resources

### PR DESCRIPTION
**Pull request by thisthat**
According to the [specification](https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/sdk-resource.md#create), Resources should have string, int64, double, or bool as value. This PR introduces the `ResourceValue` class to support this.

Also, it addresses #794 .

### CODE REVIEW

**Comment by jkwatson** [comment]
What do you think about changing the API's `AttributeValue` class to be named `TypedValue`, and re-using it for this use-case? If we want to do that, I'd change this PR to just use the AttributeValue and have a follow-up to change the name.

**Comment by thisthat** [comment]
I agree, we already have a lot of classes in the project ;) 
Addressed in 19bc9bb

**Comment by bogdandrutu** [comment]
So we are going to rename the `AttributeValue` class and move it to common package in a separate PR?

**Comment by carlosalberto** [comment]
Shall we add a `api/src/main/java/io/opentelemetry/common` package (in the aforementioned follow-up PR)?

I personally prefer the current name, btw ;)

**Comment by bogdandrutu** [comment]
@carlosalberto so you prefer the name but want to change the package correct?

**Comment by carlosalberto** [comment]
@bogdandrutu Yes, that my very personal take on this.

File: sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/Ec2Resource.java [hunk 1]
```diff
@@ -46,19 +47,29 @@ public static Resource getResource() {
   // This can be tested now with a fake info and host.
   static Resource getResourceFromInfoAndHost(
       @Nullable InstanceInfo info, @Nullable String hostname) {
-    Map<String, String> labels = new HashMap<>();
-    labels.put(ResourceConstants.CLOUD_PROVIDER, CLOUD_PROVIDER_AWS);
+    Map<String, AttributeValue> labels = new HashMap<>();
+    labels.put(
+        ResourceConstants.CLOUD_PROVIDER, AttributeValue.stringAttributeValue(CLOUD_PROVIDER_AWS));
```
Comment (bogdandrutu) [inline sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/Ec2Resource.java]: Make CLOUD_PROVIDER_AWS an AttributeValue, also everywhere where we have constants for values.

Comment (thisthat) [reply sdk_contrib/aws_v1_support/src/main/java/io/opentelemetry/sdk/contrib/trace/aws/Ec2Resource.java]: Beside in testing code, this is the only constant resource value we have! 
Done in 173c59d

**Pull request merged**